tests/test_datasets/test_dataset_functions.py

```diff
@@ -1331,3 +1332,44 @@ def test_get_dataset_cache_format_feather(self):
         self.assertEqual(X.shape, (150, 5))
         self.assertEqual(len(categorical), X.shape[1])
         self.assertEqual(len(attribute_names), X.shape[1])
+
+    def test_data_edit(self):
+
+        # admin key for test server (only admins or owners can edit datasets).
+        # all users can edit their own datasets)
+        openml.config.apikey = "d488d8afd93b32331cf6ea9d7003d4c3"
+
+        # case 1, existing version edit
+        did = 564
+        result = edit_dataset(
+            did,
```

Comment:
For each case, please add a comment which fields trigger whether or not they are edits or new versions.
Maybe iterate over all the attributes in a loop so that you can test each attribute separately for the "new version" one? Though if it increases the unit test runtime too much, I don't think that would be worth it.